thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -178,16 +193,19 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
         if (!inputPrepared)
         {
             inputPrepared = true;
+            activeOutputCount = initActiveOutputCount;
             PARENT::start();
-            remainingOutputs = activeOutputs;
             ForEachItemIn(o, outputs)
             {
                 CSplitterOutput *output = (CSplitterOutput *)outputs.item(o);
-                if (output && output->isStopped())
-                    --remainingOutputs;
+                if (output->isStopped())
```

Comment:
Actually the race may be worse than the old code.  I think in the code code smartBuffer->queryOutput(n)->stop() might have been called more than once.  With the new code it could stop prematurely.